### ISSUE TITLE
Webmin Software Packages 'Package is no longer installed' (v1.760/v1.770 on v14.0

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#493]
Initially I thought that this issue was related to #492. There are some Webmin issues after migrating from v13.0 to v14.0 (using TKLBAM). Jacob has done great work troubleshooting (see the [forums](http://www.turnkeylinux.org/forum/support/20151001/file-server-appliance14-webmin-not-working)); but there still remains an issue.

Namely missing functionality:

> system - software packages, top right button 'package tree', expand first line A-E, the first 10 or so packages I click on, all give the 'no longer installed' message.
##

**Comment by JedMeister** [comment]
Someone else has reported: https://virtualmin.com/node/38076

**Comment by JedMeister** [comment]
Workaround:
````
apt-get install dselect && dselect update
````

**Comment by JedMeister** [comment]
The resolution to this issue will be to add `dselect` as a dendency of the `webmin-software` package.

**Comment by JedMeister** [comment]
We have changed tack on this issue. I personally believe it is a problem with Webmin. There is ongoing discussion with Webmin upstream. Hopefully Jamie (Webmin lead dev) will accept my rationale and suggested fix so this issue will resolve itself with the next Webmin release.

FYI the discussion thread is [here](https://sourceforge.net/p/webadmin/bugs/4676/)

So I will leave this bug open for now, but retarget to v14.2 and make a note that it's essentially an upstream issue.

**Comment by JedMeister** [comment]
Also for the record, if you want to use the above but want the package lists to be auto updated whenever you run `apt-get update` then copy paste the following into your appliance commandline:
````
cat > /etc/apt/apt.conf.d/05update-dpkg << EOF
APT::Update::Post-Invoke {"dpkg --clear-avail; echo -n 'Updating dpkg package lists'; for LIST in $(ls /var/lib/apt/lists/*Packages); do echo -n '.'; dpkg --merge-avail $LIST >/dev/null; done; echo"};
EOF
````
The other option is to run `dselect update` from time to time (or even put it in a cron job if you want).

**Comment by JedMeister** [comment]
This issue has been [fixed upstream in Webmin](https://sourceforge.net/p/webadmin/bugs/4676/#fcae) so will get included when we update Webmin.

**Comment by JedMeister** [comment]
closed with the release of v1.831 packages (and day now...)
